### PR TITLE
Recursive Test Discovery

### PR DESCRIPTION
**Pull request by apahim**
- Recursion, when engaged (using `:avocado: recursive`), will discover all parents up to the base class.
- `:avocado: enable` works as usual, but it's not needed in a recursive discovery.
- `:avocado: disable` works as usual, but will abort the recursive discovery from that point.

### CODE REVIEW

**Comment by apahim** [comment]
[Inherit `test*` methods from parents](https://trello.com/c/GtFlhcOj/1011-inherit-%60test*%60-methods-from-parents)

File: docs/source/WritingTests.rst [hunk 1]
```diff
@@ -1251,6 +1251,70 @@ The docstring ``:avocado: disable`` is evaluated first by Avocado,
 meaning that if both ``:avocado: disable`` and ``:avocado: enable`` are
 present in the same docstring, the test will not be listed.
 
+
+Recursively Discovering Tests
+-----------------------------
+
+In addition to the ``:avocado: enable`` and ``:avocado: disable``
+docstring directives, Avocado has support to the ``:avocado: recursive``
```
Comment (jnmnus) [inline docs/source/WritingTests.rst]: s/support to/support for/

File: docs/source/WritingTests.rst [hunk 2]
```diff
@@ -1251,6 +1251,70 @@ The docstring ``:avocado: disable`` is evaluated first by Avocado,
...
+directive. It is intended to be used in inherited classes when you want
+to tell Avocado to also discover the inherited class parents.
```
Comment (jnmnus) [inline docs/source/WritingTests.rst]: s/inherited class parents/the ancestor classes/

File: docs/source/WritingTests.rst [hunk 3]
```diff
@@ -1251,6 +1251,70 @@ The docstring ``:avocado: disable`` is evaluated first by Avocado,
...
+
+The ``:avocado: recursive`` directive will make Avocado to evaluate all
```
Comment (jnmnus) [inline docs/source/WritingTests.rst]: s/make/direct/

File: docs/source/WritingTests.rst [hunk 4]
```diff
@@ -1251,6 +1251,70 @@ The docstring ``:avocado: disable`` is evaluated first by Avocado,
...
+the parents until the base class, the one which inherits from
```
Comment (jnmnus) [inline docs/source/WritingTests.rst]: s/parents/ancestors of the class/

**Comment by apahim** [comment]
Thank you @jnmnus. V2 will address your comments.

**Pull request closed without merging**